pkg/scheduler/dependency_graph.go

```diff
@@ -281,6 +319,73 @@ func unique(slice []string) []string {
 	return result
 }
 
+type ReplicaList client.ReplicaList
+
+func (rl *ReplicaList) Len() int {
+	return len(rl.Items)
+}
+
+func (rl *ReplicaList) Less(i, j int) bool {
+	if rl.Items[i].CreationTimestamp.Equal(rl.Items[j].CreationTimestamp) {
+		return rl.Items[i].UID < rl.Items[j].UID
+	}
+	return rl.Items[i].CreationTimestamp.Before(rl.Items[j].CreationTimestamp)
+}
+
+func (rl *ReplicaList) Swap(i, j int) {
+	rl.Items[i], rl.Items[j] = rl.Items[j], rl.Items[i]
+}
+
+func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.DependencyGraphOptions) ([]string, error) {
```

Comment:
Yes, I'm going to write unit tests for this and many other functions, just later in separate PRs